### ISSUE TITLE
[PCollections] jackson-datatype-pcollections is missing serializers

### ISSUE THREAD

**Issue by mateusz-bajorek** [FasterXML/jackson-datatypes-collections#35]
As in the topic, PCollections support in jackson is missing serializers for `TreePVector`, `MapPSet` and others.
From the module class:
```java
@Override
public void setupModule(SetupContext context)
{
    context.addDeserializers(new PCollectionsDeserializers());
}
```

**Comment by cowtowncoder** [comment]
Indeed. If anyone is interested in adding support, would love to help get PRs merged.

**Comment by vanyamanko** [comment]
Hi @cowtowncoder! Serialization of PCollections types already works via the standard JDK collection/map serializers (see `TestPCollections.withoutSerializers()`), so dedicated serializers don't seem necessary.

What's actually missing are deserializers for types added in newer PCollections versions: `TreePSet`/`PSortedSet`, `TreePMap`/`PSortedMap`, `OrderedPMap` and `AmortizedPQueue`/`PQueue`. Reading JSON into any of these currently fails with "Cannot construct instance of...".

I've opened #229 that adds them.
